src/ecpool_worker.erl

```diff
@@ -74,17 +74,17 @@ start_link(Pool, Id, Mod, Opts) ->
 client(Pid) ->
     gen_server:call(Pid, client, infinity).
 
--spec(exec(pid(), action()) -> Result :: any() | {error, Reason :: term()}).
-exec(Pid, Action) ->
-    gen_server:call(Pid, {exec, Action}, infinity).
+-spec(exec(pid(), action(), timeout()) -> Result :: any() | {error, Reason :: term()}).
+exec(Pid, Action, Timeout) ->
+    gen_server:call(Pid, {exec, Action}, Timeout).
 
 -spec(exec_async(pid(), action()) -> Result :: any() | {error, Reason :: term()}).
 exec_async(Pid, Action) ->
-    gen_server:call(Pid, {exec_async, Action}).
+    gen_server:cast(Pid, {exec_async, Action}).
 
 -spec(exec_async(pid(), action(), callback()) -> Result :: any() | {error, Reason :: term()}).
```

Comment:
gen_server:cast always return ok.
why do we change this ?

Reply:
never mind the question.
now I see that the handle_call clause always return `ok`

Reply:
actually, `gen_server:call` will crash with `noproc` when the callee process exits. 
it also has a pushback when the process has too many message in mailbox.
is `gen_server:cast` really ok here ?

Reply:
I am not sure if `gen_server:cast` is OK for us, it is probably flood the worker and should never be used. I changed it to `gen_server:cast` here because it should be `cast`, as the API is named `*_async`, and `gen_server:reply` is not really `async`.

Reply:
'async' means the call is returned before the actual result is sent back.
in this sense, the original behaviour is indeed async.

Reply:
But it is only async for the first time the caller calls the API. If the worker is working on a task, the caller will be blocked and act as a `sync` call.

So I change it to a real async call using `cast`.

Reply:
Synced or not really depends on how the callback is implemented.
but ok, I got your points.